Userland/Applications/PixelPaint/Tools/GradientTool.h

```diff
@@ -32,19 +32,42 @@ class GradientTool : public Tool {
     virtual StringView tool_name() const override { return "Gradient Tool"sv; }
 
 private:
+    enum class GradientMode {
+        Linear,
+        Radial,
+        __Count,
+    };
+
+    enum class IconStyle {
+        None,
+        ChangeWidthAndAngle,
+        RadialWith,
+    };
+
+    enum class TranslationType {
+        Move,
+        Rotate,
+    };
```

Comment:
thatʼs a misnomer too. the main issue with that name is that translation is a move (or vice versa) but rotation could be hardly considered as such (in euclidean plane setting). should be more of `Transformation` or `TransformationType` with `Move`/`Translate` and `Filp`/`Rotate` items (or “translation type” having “plain”/“simple” and “flipped”/“inverted”/“reflected”/“mirrored”, optionally “with filp/inversion/reflection”).

in fact youʼre making a translation followed possibly by a point/central reflection/inversion (or axial). i believe thereʼs something suspicious in it—thereʼre three use cases of `translate_gradient_points` with one of them having point reflection applied upfront… not sure about it (the order of application might be the thing here), but maybe you could remove that reflection in the call site and set the second parameter to `Move`, so in the end you would not need it at all?

not sure what exactly is happening inside from reading so cannot be of much more help here… 😞

Reply:
Thanks for the recommendations ;-)
I decided to go with a mix of them... `Transformation(MoveAllPoints, MoveWithInversionPoints)`

Regards

Reply:
“move with inversion points” is still suspicious. maybe i could help you without a need to read the code diligently or debugging carefully, though: could you describe what this option exactly controls (in your own words)?

Reply:
Hey @konradekk,
let me try to clarify my thoughts...
- MoveAllPoints is used when dragging the whole gradient so all points are modiefied with the delta
- MoveWithInversionPoints is used when the gradient shape is change e.g. linear gradient changes width or radial gradient changes orientation 

So my initial thought was to use "Rotate" as the gradient points "kind of" rotate arround the center when the gradient is modified.

Maybe a better name for MoveWithInversionPoints would be "ApplyShapeChanges"?

Reply:
it seems that youʼre changing the coordinate(s space)… the first one is related to the global (i.e. canvas, no change), and the second one is related to the local (i.e. shape, polar?!). 🤔

those are just guesses—it would be nice to name properly what we do here exactly. 😓
(reading the code in seemingly inevitable!)

Reply:
i changed the methods to that there is now one method that is responsible to move all the points when dragging the whole gradient and one method that is responsible for managing the points when adusting the angle of the gradient.